website/docs/r/kubernetes_cluster.html.markdown

```diff
@@ -217,19 +217,19 @@ The following arguments are supported:
 
 `network_profile` supports the following:
 
-* `network_plugin` - (Required) Network plugin to use for networking. Currently supported values are 'azure' and 'kubenet'. Changing this forces a new resource to be created.
+* `network_plugin` - (Required) Network plugin to use for networking. Currently supported values are `azure` and `kubenet`. Changing this forces a new resource to be created.
 
 -> **NOTE:** When `network_plugin` is set to `azure` - the `vnet_subnet_id` field in the `agent_pool_profile` block must be set.
 
-* `service_cidr` - (Optional) The Network Range used by the Kubernetes service. This is required when `network_plugin` is set to `kubenet`. Changing this forces a new resource to be created.
+* `service_cidr` - (Optional) The Network Range used by the Kubernetes service. Changing this forces a new resource to be created.
 
 ~> **NOTE:** This range should not be used by any network element on or connected to this VNet. Service address CIDR must be smaller than /12.
 
-* `dns_service_ip` - (Optional) IP address within the Kubernetes service address range that will be used by cluster service discovery (kube-dns). This is required when `network_plugin` is set to `kubenet`. Changing this forces a new resource to be created.
+* `dns_service_ip` - (Optional) IP address within the Kubernetes service address range that will be used by cluster service discovery (kube-dns). Changing this forces a new resource to be created.
 
-* `docker_bridge_cidr` - (Optional) IP address (in CIDR notation) used as the Docker bridge IP address on nodes. This is required when `network_plugin` is set to `kubenet`. Changing this forces a new resource to be created.
```

Comment:
we should add in "This field can only be set when `network_plugin` is set to `kubenet`"

Reply:
Updated.

---
In reply to: [207462137](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r207462137) [](ancestors = 207462137)

Reply:
It should also be noted that the docker_bridge_cidr must have the last octet noted or the RP will fail. e.g 172.26.0.1/16 not 172.26.0.0/16 The server side logic is testing for this last octet and will stop the deployment.

Also the logic above for dns_service_ip is incorrect. It is only required is serviceCidr is entered not docker_bride_cidr

Reply:
We tried this combination:
```hcl
network_profile {
    network_plugin     = "kubenet"
    dns_service_ip     = "10.10.0.10"
    docker_bridge_cidr = ""
    service_cidr       = "10.10.0.0/16"
}
```
It still fails due to not all of 3 are set.

---
In reply to: [207924196](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r207924196) [](ancestors = 207924196)

Reply:
For the octet, I don't quite understand, can you be more specific on the documentation change?

---
In reply to: [208059432](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r208059432) [](ancestors = 208059432,207924196)

Reply:
If someone enters what they normally think of a CIDR address is such as 172.16.0.0/16 the RP will kick back an error saying that docker_bridge_Cidr is not a valid cidr address. It must be the first available address in the space and the prefix number, not the network name and prefix.

Reply:
Where can we find the these rules? I feel we probably can provide a link to how to define the addresses here instead. Anyway, feel free to send another PR to update this document, that will be more efficient.

---
In reply to: [208078669](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r208078669) [](ancestors = 208078669)

Reply:
Let's take it offline.

---
In reply to: [208059738](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r208059738) [](ancestors = 208059738,208059432,207924196)